ambry-account/src/main/java/com/github/ambry/account/AccountUpdater.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.commons.HelixNotifier;
+import com.github.ambry.commons.Notifier;
+import com.github.ambry.config.HelixPropertyStoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.Utils;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Properties;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import static com.github.ambry.account.AccountUtils.*;
+
+
+/**
+ * A component that is used to create/update {@link Account}s to {@code ZooKeeper} server. Underneath it
+ * employs an {@link AccountService} to do the actual write operation, and a {@link Notifier} to publish
+ * message for account metadata change once the write operation is completed.
+ */
+class AccountUpdater {
```

Comment:
For now it is only used by the `AmbryUpdateTool`. Should I still rename `AmbryUpdater`?